src/main/haskell/kore/test/Test/Kore/Step/Step.hs

```diff
@@ -260,6 +277,218 @@ test_simpleStrategy =
             =<< actualZeroStepLimit)
     ]
 
+test_onePathStrategy :: [TestTree]
+test_onePathStrategy = give symbolOrAliasSorts
+    [ testCase "Runs zero steps" $ do
+        -- Removal axiom: a => bottom
+        -- Coinductive axiom: a => b
+        -- Normal axiom: a => c
+        -- Start pattern: a
+        -- Expected: a
+        [(actual, _proof)] <- runOnePathSteps
+            metadataTools
+            (Limit 0)
+            (ExpandedPattern.fromPurePattern Mock.a)
+            (simpleAxiom Mock.a mkBottom)
+            [simpleAxiom Mock.a Mock.b]
+            [simpleAxiom Mock.a Mock.c]
+        assertEqualWithExplanation ""
+            (ExpandedPattern.fromPurePattern Mock.a)
+            actual
+    , testCase "Axiom priority, first step" $ do
+        -- Removal axiom: a => bottom
+        -- Coinductive axiom: a => b
+        -- Normal axiom: a => c
+        -- Start pattern: a
+        -- Expected: a, since a->bottom and bottom is ignored
+        [(_actual, _proof)] <- runOnePathSteps
+            metadataTools
+            (Limit 1)
+            (ExpandedPattern.fromPurePattern Mock.a)
+            (simpleAxiom Mock.a mkBottom)
+            [simpleAxiom Mock.a Mock.b]
+            [simpleAxiom Mock.a Mock.c]
+        assertEqualWithExplanation ""
+            (ExpandedPattern.fromPurePattern Mock.a)
+            _actual
+        -- Removal axiom: d => bottom
+        -- Coinductive axiom: a => b
+        -- Normal axiom: a => c
+        -- Start pattern: a
+        -- Expected: c, since coinductive axioms are applied only at the second
+        -- step
+        [(_actual, _proof)] <- runOnePathSteps
+            metadataTools
+            (Limit 1)
+            (ExpandedPattern.fromPurePattern Mock.a)
+            (simpleAxiom Mock.d mkBottom)
+            [simpleAxiom Mock.a Mock.b]
+            [simpleAxiom Mock.a Mock.c]
+        assertEqualWithExplanation ""
+            (ExpandedPattern.fromPurePattern Mock.c)
+            _actual
+    , testCase "Axiom priority, second step" $ do
+        -- Removal axiom: b => bottom
+        -- Coinductive axiom: b => c
+        -- Normal axiom: b => d
+        -- Normal axiom: a => b
+        -- Start pattern: a
+        -- Expected: b, since a->b->bottom and bottom is ignored
+        [(_actual, _proof)] <- runOnePathSteps
```

Comment:
I would expect bottom here, as well, as after `a -> b` you can apply `b -> bottom`